pkg/services/did.go

```diff
@@ -128,6 +134,30 @@ func (d *DidDocumentServices) GetDidDocument(ctx context.Context, did string, op
 		},
 	)
 
+	walletAddress, err := resolver.WalletAddress()
+
+	if err == nil && opts.Signature != "" {
+		primaryType := IdentityStateType
```

Comment:
I'm not clear why do you separate something by request type.
I should get genesis state and identity state if I pass state param. Why do we need this primaryType?

Reply:
In the case of signature as I pointed we only get one of GlobalState or IdentityState to be signed.